sources/api/corndog/src/main.rs

```diff
@@ -45,20 +51,30 @@ fn run() -> Result<()> {
     SimpleLogger::init(args.log_level, LogConfig::default()).context(error::LoggerSnafu)?;
 
     // If the user has kernel settings, apply them.
-    let kernel = get_kernel_settings(args.config_path)?;
     match args.subcommand.as_ref() {
         "sysctl" => {
+            let kernel = get_kernel_settings(args.config_path)?;
             if let Some(sysctls) = kernel.sysctl {
                 debug!("Applying sysctls: {:#?}", sysctls);
                 set_sysctls(sysctls);
             }
         }
         "lockdown" => {
+            let kernel = get_kernel_settings(args.config_path)?;
             if let Some(lockdown) = kernel.lockdown {
                 debug!("Setting lockdown: {:#?}", lockdown);
                 set_lockdown(&lockdown)?;
             }
         }
+        "generate-hugepages-setting" => {
+            let hugepages_setting = generate_hugepages_setting()?;
+            // We will only fail if we cannot serialize the output to JSON string.
+            // sundog expects JSON-serialized output so that many types can be represented, allowing the
+            // API model to use more accurate types.
```

Comment:
```suggestion
            let hugepages_setting = generate_hugepages_setting();
            // We will only fail if we cannot serialize the output to JSON string.
            // sundog expects JSON-serialized output so that many types can be represented, allowing the
            // API model to use more accurate types.

```

I think this comment can't be true because `generate_hugepages_setting()` returns a `Result`. Can we either make it not return a `Result`, or else add code at this level which explicitly performs the fallback behavior we're expecting if generating hugepage settings fails?

I'd also add commentary here about what the fallback behavior is. Right now, you have to dive into `generate_hugepages_setting()` to find out.